### DISCUSSION TITLE
Using react-markdown with links

### DISCUSSION THREAD

**Discussion by Nel-Q** [remarkjs/remark#1025] [Q&A] [answered]
Can I use the component with a link and a filename from local storage? I am trying to get my files from AWS and wondering if I can just use a link and not have to download the files to my local storage.

**Comment by wooorm** [comment, marked as answer]
This is super vague. Please look at the support guide and improve your question to include much more information.

**Reply by Nel-Q** [reply]
Let me clarify my question. The example of how to use react-markdown is either to declare the markdown like this 
`const markdown = `Just a link: https://reactjs.com``.  or fetch the file from a local storage like `import README from ./markdown/README.md`. I was wondering if there is a way to just fetch the url of a markdown file like this `const markdown = https://github/Repo/README.md`

**Reply by wooorm** [reply]
1. You‘re making some markdown mistakes, see https://commonmark.org/help/ for how it works, it can help bring your question across
2. You’re mixing up terms, local storage is https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API. Here you mean the file system? Your bundler?
3. You mention fetching things. There is a thing called fetch on the web. You can use it to fetch things from URLs. https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API.

**Reply by Nel-Q** [reply]
Thank you for your help and your prompt response. Let me just share a snippet of my code maybe that might help. I was able to use react-markdown using the following lines of code:
```
import * as React from "react";
import ReactMarkdown from 'react-markdown';
import README from "./markdown/README.md"

const HighlightFeature = () => {
    let [content, setContent] = React.useState({md: ""});

    async function fetchData() {
        fetch(README)
        .then((res) => res.text())
        .then((md) => {
            setContent({ md })
        })
      }

    React.useEffect(()=> {
        fetchData()
    }, [])

    return (
      <div className="post">
        <ReactMarkdown children={content.md}/>
      </div>
    )
  }
```

But when I switch to the following lines, it does not work anymore:
```
    const HighlightFeature = () => {
    let [content, setContent] = React.useState({md: ""});

    async function fetchData() {
        fetch('https://github.com/Nel-Q/Asteroid-Escape/blob/main/README.md')
        .then((res) => res.text())
        .then((md) => {
            setContent({ md })
        })
      }

    React.useEffect(()=> {
        fetchData()
    }, [])

    return (
      <div className="post">
        <ReactMarkdown children={content.md}/>
      </div>
    )
  }
```

**Reply by wooorm** [reply]
You have problems using React asynchronously. There are other resources on how to use React that are much better at that than I am.

**Reply by ChristianMurphy** [reply]
@Nel-Q as @wooorm notes this isn't really `react-markdown` related, but JavaScript and React.
Some things to consider, in your examples:
1. you are using `async` but not `await`, either use `await` instead of `then` or drop `async`.
2. you aren't `.catch`ing any potential errors, the difference between the first and second example is the URL, I suspect there is a network error, and `.catch` and logging the error would allow you to debug that.

**Comment by Nel-Q** [comment]
So you are saying react-markdown supports fetching like that and my async function is the one messing things up?

**Reply by wooorm** [reply]
React supports async.
`react-markdown` has nothing to do with that.
You need to configure React properly.

**Reply by Nel-Q** [reply]
Thank you for all the help. I figure out what the problem was and it wasn't related to `react-markdown`
